src/index.ts

```diff
@@ -13,23 +13,48 @@ import * as process from "process";
 
 import routes from "./routes";
 import consts from "./consts";
+import { listAvailableReleases } from "./replicated/release";
 
 // TODO yargs
 function validate() {
-  let validThresholds = ["info", "warn", "error", "off"];
-  let thesholdInvalid = _.indexOf(validThresholds, consts.lintThreshold) === -1;
+  const validThresholds = ["info", "warn", "error", "off"];
+  const thesholdInvalid = _.indexOf(validThresholds, consts.lintThreshold) === -1;
   if (thesholdInvalid) {
     console.log(chalk.red(`unknown lint threshold ${consts.lintThreshold}. Choices are ${validThresholds.join(", ")}`));
     process.exit(1);
   }
 }
 
-function watch() {
-  if (!fs.existsSync(consts.localPath)) {
-    console.log(chalk.red(`Please create your application yaml at ${consts.localPath} before continuing.`));
-    process.exit(1);
+export function watch() {
+  if (!fs.existsSync(path.join(consts.localPath, "current.yaml"))) {
+    console.log(chalk.red(`No file detected at ${path.join(consts.localPath, "current.yaml")}. An attempt will be made to create one for you when Replicated first requests it.`));
+    return;
   }
-  console.log(chalk.green(`Watching "${consts.localPath}" for changes. Any updates to this file will be sent as an application update.`));
+  console.log(chalk.green(`Watching "${path.join(consts.localPath, "current.yaml")}" for changes. Any updates to this file will be sent as an application update.`));
+  fs.watchFile(path.join(consts.localPath, "current.yaml"), (curr, prev) => {
+    const releases = listAvailableReleases();
+    if (releases.length === 0) {
+      fs.copyFileSync(path.join(consts.localPath, "current.yaml"), path.join(consts.localPath, "releases", "1.yaml"));
+      console.log("Created first release at releases/1.yaml");
+    } else {
+      // next release is at the number of last, plus 1
+      const last = releases[releases.length - 1];
+      let name = "";
+      if (last.indexOf(".yaml") !== -1) {
```

Comment:
I'd split this on `.` and and take first part as `name`

Reply:
Yeah, that sounds like a better idea